src/core/SurfaceLayer.cpp

```diff
@@ -14,47 +11,44 @@ void ngen::SurfaceLayer::update_models(boost::span<double> catchment_outflows,
     
     Layer::update_models(catchment_outflows, catchment_indexes, nexus_downstream_flows, nexus_indexes, current_step);
 
-    //At this point, could make an internal routing pass, extracting flows from nexuses and routing
-    //across the flowpath to the next nexus.
-    //Once everything is updated for this timestep, dump the nexus output
+    // Once contributing catchments are updated for this timestep, dump the nexus output
     for(const auto& id : features.nexuses()) 
     {
         std::string current_timestamp = simulation_time.get_timestamp(current_time_index);
         
         #if NGEN_WITH_MPI
-        if (!features.is_remote_sender_nexus(id)) { //Ensures only one side of the dual sided remote nexus actually doing this...
+        // Ensures only one side of the dual sided remote nexus actually does this
+        if (features.is_remote_sender_nexus(id)) continue;
         #endif
 
-        //Get the correct "requesting" id for downstream_flow
+        // Get the correct "requesting" id for downstream_flow
         const auto& nexus = features.nexus_at(id);
         const auto& cat_ids = nexus->get_receiving_catchments();
+
+        if (cat_ids.size() > 1) {
+            std::string error = "Nexus '" + id + "' violates dendritic hydrofabric network assumption";
+            LOG(error, LogLevel::FATAL);
+            throw std::runtime_error(error);
+        }
+
         std::string cat_id;
-        if( cat_ids.size() > 0 ) {
-            //Assumes dendridic, e.g. only a single downstream...it will consume 100%  of the available flow
+        if( cat_ids.size() == 1 ) {
+            // With a dendritic network, there can only be a single downstream. It will consume 100%  of the available flow
             cat_id = cat_ids[0];
         }
         else {
-            //This is a terminal node, SHOULDN'T be remote, so ID shouldn't matter too much
+            // This is a terminal node, SHOULDN'T be remote, so ID shouldn't matter too much
             cat_id = "terminal";
         }
 
         //std::cerr << "Requesting water from nexus, id = " << id << " at time = " <<current_time_index << ",  percent = 100, destination = " << cat_id << std::endl;
         double contribution_at_t = features.nexus_at(id)->get_downstream_flow(cat_id, current_time_index, 100.0);
+
 #if NGEN_WITH_ROUTING
         int nexus_index = nexus_indexes[id];
         nexus_downstream_flows[nexus_index] += contribution_at_t;
 #endif // NGEN_WITH_ROUTING
-        if(nexus_outfiles[id].is_open()) {
```

Comment:
I think it may be premature to remove the nexus file outputs. Currently, they take an insignificant amount of the total runtime to write, so it's not acting a bottleneck at the moment. I'm also worried that there are lots of other parts of the project that are quietly relying on those files existing, so this change should be in coordination with all parts. For instance, cal manager would fail [here](https://github.com/NGWPC/nwm-cal-mgr/blob/37724ba7dbc9884844a9296fb2cf9de6c7cce4ac/python/calib/src/calib/calibration_set.py#L139) if no nexus files were generated.